pybamm/util.py

```diff
@@ -345,3 +345,21 @@ def install_jax(arguments=None):  # pragma: no cover
             f"jaxlib>={JAXLIB_VERSION}",
         ]
     )
+
+
+def have_optional_dependency(module_name, attribute=None):
+    try:
+        module = importlib.import_module(module_name)
+        if attribute:
+            if hasattr(module, attribute):
+                imported_attribute = getattr(module, attribute)
+                return imported_attribute
+            else:
+                raise ImportError(f"Optional dependency {module_name}.{attribute} is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details.")
+        else:
+            return module
+    except ImportError:
+        if attribute:
+            raise ImportError(f"Optional dependency {module_name}.{attribute} is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details.")
```

Comment:
```suggestion
            raise ImportError(f"Optional dependency {module_name} is not available. See https://docs.pybamm.org/en/latest/source/user_guide/installation/index.html#optional-dependencies for more details.")
```

I feel users don't need to know the submodule we are using from a particular optional dependency, I feel that just the name of the dependency in the error message should be enough information.

Also, is it possible to directly embed the installation command in the error message? Like, is it possible to programmatically access the optional dependencies from `setup.py` and soon `pyproject.toml`? We can then ask users to install `pip install pybamm[cite]` if `pybtex` isn't found, and similarly for the other dependencies. Either way, the link will be descriptive enough so we can skip doing this if it isn't feasible without the addition of an extra dictionary (it just becomes another thing we need to maintain otherwise).